### PR TITLE
Expose API to change form field background colour

### PR DESCRIPTION
**Pull request by darrenchann**
setFormFieldHighlightColor API

### CODE REVIEW

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]
convert rgb to color should have util similar to ios, and use similar approach like other method i.e. 255F instead of 255.

**Comment by dluco** [comment]
There should be 4 color components too, for RGBA. Needs an `alpha` entry from JS.

File: API.md [hunk 1]
```diff
@@ -3945,3 +3945,16 @@ Requires followSystemDarkMode={false} in order for functionality of this prop
...
 ```
+
+#### setFormFieldHighlightColor
+Sets the form field's highlight color on the PDF.
+
+Parameters:
+
+Name | Type | Description
+--- | --- | ---
+fieldHightlightColor | object | the form field highlight color, in the format `{red: number, green: number, blue: number}`, each number in range [0, 255]
```
Comment (dluco) [inline API.md:3956]:
```suggestion
fieldHightlightColor | object | the form field highlight color, in the format `{red: number, green: number, blue: number, alpha: number}`, each number in range [0, 255]
```

File: API.md [hunk 2]
```diff
@@ -3945,3 +3945,16 @@ Requires followSystemDarkMode={false} in order for functionality of this prop
...
+
+```js
+this._viewer.setFormFieldHighlightColor({red: 0, green: 0, blue: 255}); // blue color
```
Comment (dluco) [inline API.md:3959]:
```suggestion
this._viewer.setFormFieldHighlightColor({red: 0, green: 0, blue: 255, alpha: 128}); // transparent blue color
```

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 3]
```diff
@@ -4947,6 +4947,17 @@ public void downloadFailed(@NonNull Exception e) {
         });
     }
 
+    public void setFormFieldHighlightColor(ReadableMap fieldHighlightColor) throws PDFNetException {
+        if (getPdfViewCtrl() != null) {
+            int red = fieldHighlightColor.getInt(COLOR_RED);
+            int green = fieldHighlightColor.getInt(COLOR_GREEN);
+            int blue = fieldHighlightColor.getInt(COLOR_BLUE);
+            getPdfViewCtrl().setFieldHighlightColor(new ColorPt(red/255., green/255., blue/255.));
```
Comment (dluco) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:4955]: The `setFieldHighlightColor()` API is unusual in that it takes a 4-component `ColorPt` object, with the last component being the alpha.

(Normally a 4-component color is in the CMYK colorspace, not RGBA)

Comment (darrenchann) [reply android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:4955]: fixed

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 4]
```diff
@@ -4947,6 +4947,19 @@ public void downloadFailed(@NonNull Exception e) {
         });
     }
 
+    public void setFormFieldHighlightColor(ReadableMap fieldHighlightColor) throws PDFNetException {
+        if (getPdfViewCtrl() != null && fieldHighlightColor != null) {
+            double red = (double) (fieldHighlightColor.getInt(COLOR_RED) / 255f);
+            double green = (double) (fieldHighlightColor.getInt(COLOR_GREEN) / 255f);
+            double blue = (double) (fieldHighlightColor.getInt(COLOR_BLUE) / 255f);
+            double alpha = (double) (fieldHighlightColor.getInt(COLOR_ALPHA) / 255f);
+            ColorPt colorPt = new ColorPt(red, green, blue, alpha);
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:4956]: ideally you have a `convertRGBAToColorPt` method where you can reuse it in various places

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java (hunk 4 above)
Comment (darrenchann) [reply android/src/main/java/com/pdftron/reactnative/views/DocumentView.java]: new method convertRGBAToColorPt

**Review by sgong-pdftron — APPROVED** [review]

**Pull request merged**